tasks/quality_gates.py

```diff
@@ -210,19 +210,22 @@ def parse_and_trigger_gates(ctx, config_path=GATE_CONFIG_PATH):
 
     metric_handler.send_metrics_to_datadog()
 
-    metric_handler.generate_metric_reports(ctx, branch=branch, is_nightly=nightly_run)
-
     # We don't need a PR notification nor gate failures on release branches
     if not is_a_release_branch(ctx, branch):
         github = GithubAPI()
         if github.get_pr_for_branch(branch).totalCount > 0:
             ancestor = get_common_ancestor(ctx, "HEAD")
-            metric_handler.generate_relative_size(ctx, ancestor=ancestor)
+            metric_handler.generate_relative_size(
+                ctx, ancestor=ancestor, report_path="ancestor_static_gate_report.json"
+            )
```

Comment:
Question: Given that we're generating the reports regardless of the `is_a_release_branch` condition, shouldn't we be doing this bit also in all cases? Though I don't have the full picture of what these "metric reports" might be used for.